clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -324,6 +325,21 @@ void SYCL::fpga::BackendCompiler::ConstructJob(
       FPGADepFiles.push_back(II);
     else
       CmdArgs.push_back(C.getArgs().MakeArgString(Filename));
+    // Check for any AOCR input, if found use that as the project report name
+    StringRef Ext(llvm::sys::path::extension(Filename));
+    if (Ext.empty())
+      continue;
+    if (getToolChain().LookupTypeForExtension(Ext.drop_front()) ==
+        types::TY_FPGA_AOCR) {
+      // Keep the base of the .aocr file name.  Input file is a temporary,
+      // so we are stripping off the additional naming information for a
+      // cleaner name.
+      SmallString<128> NameBase(Filename.substr(0, Filename.length() - 12));
+      NameBase.append(".aocr");
+      llvm::sys::path::replace_extension(NameBase, "prj");
```

Comment:
Is there a reason not to prefer the following?
```suggestion
      NameBase.append(".prj");
```

Reply:
hah - good catch.  I have streamlined based on your suggestion.